### PR TITLE
Support Wasm

### PR DESCRIPTION
**Pull request by 1l0**
Maybe this annoys you.

ref: https://njump.me/nevent1qvzqqqqqqypzq0raz2nv9acla89z2fepdaff5ymmkrewkqvtrresqqun8w2nyqf7qy88wumn8ghj77tpvf6jumt99uq3qamnwvaz7tmwdaehgu3wd4hk6tcqypzkn265slt4smjxeler3rh4kvyz5kysxtvkggk0ccmkwhcdmqznw8c0dme

### CODE REVIEW

**Comment by fiatjaf** [comment]
I always wanted to do this, thank you! So really the only part that was needed was the WebSocket connection and you just did it?

File: sdk/hints/test/libsql_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//go:build !sqlite_math_functions
+//go:build !js && !sqlite_math_functions
```
Comment (fiatjaf) [inline sdk/hints/test/libsql_test.go:1]: doesn't this mean it will compile this test when `js` is true but `sqlite_math_functions` is false?

**Comment by 1l0** [comment]
That's lovely to hear. Yeah, just as you wrote. But I'm not sure whether there could be edge cases that fail.

File: sdk/hints/test/libsql_test.go (hunk 1 above)
Comment (1l0) [reply sdk/hints/test/libsql_test.go:1]: This means `js` is false and `sqlite_math_functions` is false, I guess.

**Comment by 1l0** [comment]
And this doesn't support TinyGo unfortunately.  At least `github.com/puzpuzpuz/xsync` uses newer [RWMutex method](https://pkg.go.dev/sync#RWMutex.TryLock) which TinyGo doesn't support.

File: sdk/hints/test/libsql_test.go (hunk 1 above)
Comment (1l0) [reply sdk/hints/test/libsql_test.go:1]: It seems to be working as expected:
```sh
20:47 sdk/hints/test wasm $ GOOS=js GOARCH=wasm go test -run TestSQLiteHintsLibsql
testing: warning: no tests to run
PASS
ok      github.com/nbd-wtf/go-nostr/sdk/hints/test      1.115s
20:47 sdk/hints/test wasm $ go test -tags sqlite_math_functions -run TestSQLiteHintsLibsql
testing: warning: no tests to run
PASS
ok      github.com/nbd-wtf/go-nostr/sdk/hints/test      0.289s
20:47 sdk/hints/test wasm $ go test -run TestSQLiteHintsLibsql
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>        57245
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://ccc.org ::>        20930
    2 ::                  wss://aaa.com ::>     -1437816
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1437816
    2 ::                  wss://ccc.org ::>     -1459181
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
== relay scores for 0000000000000000000000000000000000000000000000000000000000000002
    0 ::                  wss://ccc.org ::>       125585
    1 ::                  wss://aaa.com ::>        19370
    2 ::                  wss://bbb.net ::>        19370
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
== relay scores for 0000000000000000000000000000000000000000000000000000000000000002
    0 ::                  wss://ccc.org ::>       125585
    1 ::                  wss://aaa.com ::>        19370
    2 ::                  wss://bbb.net ::>        19370
== relay scores for 0000000000000000000000000000000000000000000000000000000000000003
    0 ::                  wss://bbb.net ::>        20190
    1 ::                  wss://aaa.com ::>         4583
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
== relay scores for 0000000000000000000000000000000000000000000000000000000000000002
    0 ::                  wss://ccc.org ::>       125585
    1 ::                  wss://aaa.com ::>        19370
    2 ::                  wss://bbb.net ::>        19370
== relay scores for 0000000000000000000000000000000000000000000000000000000000000003
    0 ::                  wss://aaa.com ::>      -403324
    1 ::                  wss://bbb.net ::>     -1444050
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
== relay scores for 0000000000000000000000000000000000000000000000000000000000000002
    0 ::                  wss://ccc.org ::>       125585
    1 ::                  wss://aaa.com ::>        19370
    2 ::                  wss://bbb.net ::>        19370
== relay scores for 0000000000000000000000000000000000000000000000000000000000000003
    0 ::                  wss://aaa.com ::>      -403324
    1 ::                  wss://bbb.net ::>     -1444050
== relay scores for 0000000000000000000000000000000000000000000000000000000000000004
    0 ::                  wss://aaa.com ::>       115904
    1 ::                  wss://bbb.net ::>       110716
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
== relay scores for 0000000000000000000000000000000000000000000000000000000000000002
    0 ::                  wss://ccc.org ::>       125585
    1 ::                  wss://aaa.com ::>        19370
    2 ::                  wss://bbb.net ::>        19370
== relay scores for 0000000000000000000000000000000000000000000000000000000000000003
    0 ::                  wss://aaa.com ::>      -403324
    1 ::                  wss://bbb.net ::>     -1444050
== relay scores for 0000000000000000000000000000000000000000000000000000000000000004
    0 ::                  wss://aaa.com ::>       115904
    1 ::                  wss://bbb.net ::>       110716
    2 ::                  wss://ccc.org ::>         7446
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
== relay scores for 0000000000000000000000000000000000000000000000000000000000000002
    0 ::                  wss://ccc.org ::>       125585
    1 ::                  wss://aaa.com ::>        19370
    2 ::                  wss://bbb.net ::>        19370
== relay scores for 0000000000000000000000000000000000000000000000000000000000000003
    0 ::                  wss://aaa.com ::>      -403324
    1 ::                  wss://bbb.net ::>     -1444050
== relay scores for 0000000000000000000000000000000000000000000000000000000000000004
    0 ::                  wss://ccc.org ::>         7446
    1 ::                  wss://aaa.com ::>     -1294495
    2 ::                  wss://bbb.net ::>     -1299683
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
== relay scores for 0000000000000000000000000000000000000000000000000000000000000002
    0 ::                  wss://ccc.org ::>       125585
    1 ::                  wss://aaa.com ::>        19370
    2 ::                  wss://bbb.net ::>        19370
== relay scores for 0000000000000000000000000000000000000000000000000000000000000003
    0 ::                  wss://aaa.com ::>      -403324
    1 ::                  wss://bbb.net ::>     -1444050
== relay scores for 0000000000000000000000000000000000000000000000000000000000000004
    0 ::                  wss://aaa.com ::>       680064
    1 ::                  wss://bbb.net ::>       674876
    2 ::                  wss://ccc.org ::>         7446
= print scores
== relay scores for 0000000000000000000000000000000000000000000000000000000000000001
    0 ::                  wss://bbb.net ::>        59616
    1 ::                  wss://aaa.com ::>     -1425031
    2 ::                  wss://ccc.org ::>     -1459181
== relay scores for 0000000000000000000000000000000000000000000000000000000000000002
    0 ::                  wss://ccc.org ::>       125585
    1 ::                  wss://aaa.com ::>        19370
    2 ::                  wss://bbb.net ::>        19370
== relay scores for 0000000000000000000000000000000000000000000000000000000000000003
    0 ::                  wss://aaa.com ::>      -403324
    1 ::                  wss://bbb.net ::>     -1444050
== relay scores for 0000000000000000000000000000000000000000000000000000000000000004
    0 ::                  wss://ccc.org ::>      1516650
    1 ::                  wss://aaa.com ::>       680064
    2 ::                  wss://bbb.net ::>       674876
PASS
ok      github.com/nbd-wtf/go-nostr/sdk/hints/test      0.432s
20:48 sdk/hints/test wasm $ GOOS=js GOARCH=wasm go test -tags sqlite_math_functions -run TestSQLiteHintsLibsql
testing: warning: no tests to run
PASS
ok      github.com/nbd-wtf/go-nostr/sdk/hints/test      0.724s
```

**Comment by fiatjaf** [comment]
I can't run any of these tests. What am I doing wrong?

```
~> GOOS=js GOARCH=wasm go test -run TestSQLiteHintsLibsql
fork/exec /tmp/go-build2062213332/b001/go-nostr.test: exec format error
FAIL	github.com/nbd-wtf/go-nostr	0.004s
```

**Comment by 1l0** [comment]
Did you rename `wasmbrowsertest` to `go_js_wasm_exec`?
https://github.com/agnivade/wasmbrowsertest?tab=readme-ov-file#quickstart
